src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.cs

```diff
@@ -185,9 +181,11 @@ private static BoundNode FindNodeToAnalyze(BoundNode node)
             }
 
             /// <summary>
-            /// Optimizes local functions such that if a local function only references other local functions without closures, it itself doesn't need a closure.
+            /// The old version of <see cref="RemoveUnneededReferences"/> This is still necesary
+            /// because it modifies <see cref="CapturedVariablesByLambda"/>, which is still used
+            /// in other areas of the code. Once those uses are gone, this method should be removed.
             /// </summary>
-            private void RemoveUnneededReferences()
+            private void Old_RemoveUnneededReferences()
```

Comment:
Unfortunately, no. As the doc comment says, while this result is no longer used in the current method, it modifies the CapturedVariablesByLambda dictionary, which is used in other passes. After I've removed all uses of that dictionary I should be able to delete this.